### ISSUE TITLE
The appveyor with_basics test is flakey

### ISSUE THREAD

**Issue by lithp** [citusdata/citus#2163]
I've been re-running appveyor against https://github.com/citusdata/citus/commit/77fc0c01690d50b8c0d4e419af80619fddbc8212 (a cherry-pick of a commit which gives us better diffs onto https://github.com/citusdata/citus/commit/832c91e28c76d331b5b5b9f8acee1b4e783afe62, some commit in master from a month ago) all day to see what comes up. It's passed almost every time but [recently](https://ci.appveyor.com/project/lithp/citus/build/1.0.1198-832c91e-fix-appveyor) failed with this:

```
--- C:/projects/postgres/contrib/citus/src/test/regress/expected/with_basics.out	2018-05-09 17:09:23.344169200 -0700
+++ C:/projects/postgres/contrib/citus/src/test/regress/results/with_basics.out	2018-05-09 17:23:19.407648100 -0700
@@ -737,26 +737,26 @@
 WITH cte AS (
 	SELECT * FROM users_table
 )
 SELECT user_id, max(value_1) as value_1 FROM cte GROUP BY 1;
 WITH cte_user AS (
 	SELECT basic_view.user_id,events_table.value_2 FROM basic_view join events_table on (basic_view.user_id = events_table.user_id)
 )
 SELECT user_id, sum(value_2) FROM cte_user GROUP BY 1 ORDER BY 1, 2;
  user_id | sum  
 ---------+------
-       1 |  294
-       2 | 1026
-       3 |  782
-       4 |  943
-       5 |  806
-       6 |  220
+       1 |  336
+       2 | 1083
+       3 |  828
+       4 |  984
+       5 |  837
+       6 |  242
 (6 rows)
 
 SELECT * FROM cte_view ORDER BY 1, 2 LIMIT 5;
  user_id | value_1 
 ---------+---------
        1 |       5
        2 |       4
        3 |       5
        4 |       5
        5 |       5
```

**Comment by lithp** [comment]
The original query is:

```sql
WITH cte_user AS (                                                                                                            
    SELECT basic_view.user_id,events_table.value_2 FROM basic_view join events_table on (basic_view.user_id = events_table.user_id)                                                                     
)                                                                                                                             
SELECT user_id, sum(value_2) FROM cte_user GROUP BY 1 ORDER BY 1, 2;
```

Does this failure mean there's some kind of race condition in our CTE handling?

**Comment by onderkalaci** [comment]
Seems like already discussed in #2072. I opened the issue to see if people hit other concurrent test failures. But, I think we can simply run `with_basics ` seperately to get rid of the issue. I'll fix it today.

**Comment by marcocitus** [comment]
Fixed by #2164
